### PR TITLE
Reliability client telemetry

### PR DESCRIPTION
**Pull request by sameerag**
Extending the current `client telemetry` to include error codes. This PR scopes it only for JS-WAM as we are looking for something basic available right away. There should be a follow up PR expanding the scope to all flows and probably more data.

cc @tnorling

### CODE REVIEW

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts [hunk 1]
```diff
@@ -61,9 +61,10 @@ export class NativeInteractionClient extends BaseInteractionClient {
                 });
                 return result;
             })
-            .catch((error) => {
+            .catch((error: AuthError) => {
                 nativeATMeasurement.endMeasurement({
```
Comment (jasonnutter) [inline lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:65]: Why just this measurement?

File: lib/msal-common/src/telemetry/performance/PerformanceEvent.ts [hunk 2]
```diff
@@ -202,6 +202,13 @@ export type PerformanceEvent = {
      */
     success?: boolean | null,
 
+    /**
+     * Add specific error code in case of failure
+     *
+     * @type {string}
+     */
+    errorCode?: string,
+
```
Comment (jasonnutter) [inline lib/msal-common/src/telemetry/performance/PerformanceEvent.ts:211]: Do we want to capture suberror?

Comment (tnorling) [reply lib/msal-common/src/telemetry/performance/PerformanceEvent.ts:211]: Done

File: lib/msal-browser/src/interaction_client/NativeInteractionClient.ts (hunk 1 above)
Comment (tnorling) [reply lib/msal-browser/src/interaction_client/NativeInteractionClient.ts:65]: Added to the other failure measurements

**Review by peterzenz — APPROVED** [review]

**Review by tnorling — APPROVED** [review]

**Review by jasonnutter — APPROVED** [review]

**Comment by ghost** [comment]
:tada:`@azure/msal-browser@v2.26.0` has been released which incorporates this pull request.:tada:

Handy links:
* [Release Notes](https://github.com/AzureAD/microsoft-authentication-library-for-js/releases/tag/msal-browser-v2.26.0)
* [Official NPM Release](https://www.npmjs.com/package/@azure/msal-browser/v/2.26.0)

**Pull request merged**